ModTek/ProgressPanel.cs

```diff
@@ -52,12 +52,25 @@ public void SubmitWork(Func<IEnumerator<ProgressReport>> work)
             {
                 this.WorkList.AddLast(work);
             }
-
+            
             IEnumerator RunWorkList()
             {
                 foreach (var workFunc in WorkList) {
                     IEnumerator<ProgressReport> workEnumerator = workFunc.Invoke();
-                    while (workEnumerator.MoveNext())
+                   
+                    bool Next()
+                    {
+                        var sw = Stopwatch.StartNew();
```

Comment:
make the stopwatch instance outside the function block and use Reset() instead of StartNew() so you don't have to repay the object initialization cost on every frame